sympy/plotting/plot.py

```diff
@@ -1555,8 +1554,30 @@ def plot(*args, **kwargs):
     show = kwargs.pop('show', True)
     series = []
     plot_expr = check_arguments(args, 1, 1)
-    series = [LineOver1DRangeSeries(*arg, **kwargs) for arg in plot_expr]
 
+    new_plot = []
+    for arg in plot_expr:
+        expr = arg[0]
+        limit = arg[1]
+        frac = fraction(expr) #to find the denominator
```

Comment:
Perhaps you could use `sympy.solvers.denoms` to give you all the denominators. `fraction` is naive; I think you should use `.as_numer_denom()[1]` or `denoms`:
```python
>>> from sympy.solvers.solvers import denoms
>>> denoms(1+2/x,x)
{x}
>>> list(map(lambda e: denoms(e, x), [1+2/x,1/(3-x)]))
[{x}, {3 - x}]
```

Reply:
I used `fraction` to get the denominator as an expression so that roots can be calculated directly by using `real_roots` method. But if we use `denoms` we have to iterate over each term to calculate root. Will it be effective to do.

Reply:
`denoms` looks for all denominators. Won't a singularity anywhere cause a problem? Try your current code on the expression `1 + 2/x`.

Reply:
`1 + 2/x` fails to produce the desired result so I am replacing `fraction` with `denoms` to calculate all denominators. And will make necessary changes to code

Reply:
Doesn't the calculus module already have a singularities function?

Reply:
hmm...but it will fail for non-rational functions since, for example,
```python
>>> singularities(1/sqrt(x),x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/runner/.local/share/virtualenvs/python3/lib/python3.8/site-pac
kages/sympy/calculus/singularities.py", line 84, in singularities
    raise NotImplementedError(
NotImplementedError: Algorithms finding singularities for non-rational functions are not yet implemented.
```

Reply:
> hmm...but it will fail for non-rational functions since, for example,
> 
> ```python
> >>> singularities(1/sqrt(x),x)
> Traceback (most recent call last):
>   File "<stdin>", line 1, in <module>
>   File "/home/runner/.local/share/virtualenvs/python3/lib/python3.8/site-pac
> kages/sympy/calculus/singularities.py", line 84, in singularities
>     raise NotImplementedError(
> NotImplementedError: Algorithms finding singularities for non-rational functions are not yet implemented.
> ```

For finding the singularity we can use `denoms` and then calculate root of the each term of denominator using `real_roots`, but on applying `real_roots` function on a periodic function it is returning a single root.

Reply:
yes...`real_roots(sqrt(x - 2) - 1) -> [1]` because `real_roots` is looking at the generator of `Poly(sqrt(x - 2) -1)` which is `sqrt(x - 2)`. So it means there will be a root when `sqrt(x - 2) = 1`. It does not try to solve for the value of `x`.